components/rmrk/Gallery/History.vue

```diff
@@ -115,21 +126,19 @@ type ChartData = { buy: [Date, number][]; list: [Date, number][] };
 export default class History extends mixins(ChainMixin) {
   @Prop({ type: Array }) public events!: Interaction[];
   protected data: TableRow[] = [];
+  protected copyTableData: TableRow[] = [];
   protected priceChartData: [Date, number][][] = [];
-  protected collapsedHistory = true;
+  protected selectedEvent: string = 'all';
 
-  public mounted(): void {
-    this.collapsedHistory = true;
-
-    setTimeout(() => {
-      this.collapsedHistory = false;
-    }, 200);
+  get uniqType(): any[] {
+    return [...new Map(this.copyTableData.map(v => [v.Type, v])).values()]
```

Comment:
It would set it as the static

```ts
return [ 'MINTNFT', 'LIST',  'SEND', 'CONSUME', 'BUY']
```

Reply:
> It would set it as the static
> 
> ```ts
> return [ 'MINTNFT', 'LIST',  'SEND', 'CONSUME', 'BUY']
> ```

I wanted to show only possible filter. Like if we don't have "BUY" event in history table, I'm not showing it in select.

Reply:
So should we use in select:
- static array
- or dynamic array created from events

In the future I think we will need a global function to transform/map an array of events to Kodadot events